src/mgr/FunAction.cc

```diff
@@ -49,10 +48,12 @@ Mgr::FunAction::dump(StoreEntry* entry)
 {
     debugs(16, 5, MYNAME);
     Must(entry != nullptr);
-    if (UsingSmp())
-        storeAppendPrintf(entry, "by kid%d {\n", KidIdentifier);
+
+    openKidSection(entry, is_yaml());
+
     handler(entry);
-    if (atomic() && UsingSmp())
-        storeAppendPrintf(entry, "} by kid%d\n\n", KidIdentifier);
-}
 
+    if (atomic())
+        closeKidSection(entry, is_yaml());
+    // non-atomic() actions must call closeKidSection() when they are done
```

Comment:
Please fix non-atomic() FunAction reports to comply. In current PR code, they open but do not close `by kid` section markers. See `mgr:vm_objects` report for an example.

Reply:
I double-checked. The only non-atomic FunAction reports are:
* mgr:objects
* mgr:vm_objects
* mgr:openfd_objects

All of them are implemented via a wrapper which calls `StatObjectsStart` which calls `statObjects`  with different `filter` arguments. So covering one covers all of them

Reply:
> All of them are implemented via a wrapper which ... So covering one covers all of them

... and all of them are broken. Please fix. BTW, if you are not testing non-SMP Squids, please start testing them to reduce the number of review iterations.

Reply:
[vm_objects.txt](https://github.com/user-attachments/files/15542826/vm_objects.txt)
I do not understand. The attached output was obtained with commit ID 70e7c89b59 (current HEAD of the yaml-kids branch). I see kids section being closed.

Reply:
You are testing SMP Squid. The bug only affects non-SMP Squids.

Reply:
Now I understand. Corrected and tested.